hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/ChecksumByteBufferImpl.java

```diff
@@ -54,6 +79,23 @@
     }
   };
 
+  public static ChecksumByteBuffer crc32Impl() {
+    return new ChecksumByteBufferImpl(new CRC32());
+  }
+
+  public static ChecksumByteBuffer crc32Cimpl() {
+    if (useJava9Crc32C) {
+      try {
+        return new ChecksumByteBufferImpl(Java9Crc32CFactory.createChecksum());
```

Comment:
Should we consider a static factory here for creating the ChecksumByteBufferImpl?

Reply:
I'm not sure what you mean - the crc32Cimpl is like a static factory I think, and the code which needs to get a checksum impl calls either `checksumByteBufferImpl.crc32impl or `checksumByteBufferImpl..crc32cimpl`.

Can you post a code snippet of what you are thinking?

Reply:
What I meant was to separate out the Factory class from the interface impl to encapsulate the object creation logic separate from the behavior that the impl provides, this suggestion was based on the fact that crc32Cimpl is like a static factory method but it is co-located with the impl.